Cannot find npc id to clone 'tz_kih'

**Describe the bug**

When running the `greghib/void:2.5.0` Docker image you get the following error which crashes the server:

`Caused by: java.lang.IllegalArgumentException: Cannot find npc id to clone 'tz_kih'`

<details><summary>Stacktrace</summary>
<code>

Instance creation error : could not create instance for '[Singleton: 'world.gregs.voidps.engine.entity.character.npc.NPCs']': org.koin.core.error.InstanceCreationException: Could not create instance for '[Singleton: 'world.gregs.voidps.engine.data.definition.NPCDefinitions']'
	org.koin.core.instance.InstanceFactory.create(InstanceFactory.kt:56)
	org.koin.core.instance.SingleInstanceFactory.create(SingleInstanceFactory.kt:46)
	org.koin.core.instance.SingleInstanceFactory.get$lambda$0(SingleInstanceFactory.kt:55)
	org.koin.mp.KoinPlatformTools.synchronized(KoinPlatformTools.kt:36)
	org.koin.core.instance.SingleInstanceFactory.get(SingleInstanceFactory.kt:53)
	org.koin.core.registry.InstanceRegistry.resolveInstance$koin_core(InstanceRegistry.kt:132)
	org.koin.core.resolution.CoreResolver.resolveFromRegistry(CoreResolver.kt:87)
	org.koin.core.resolution.CoreResolver.resolveFromContextOrNull(CoreResolver.kt:74)
	org.koin.core.resolution.CoreResolver.resolveFromContextOrNull$default(CoreResolver.kt:72)
	org.koin.core.resolution.CoreResolver.resolveFromContext(CoreResolver.kt:69)
	org.koin.core.scope.Scope.resolveFromContext(Scope.kt:321)
	org.koin.core.scope.Scope.stackParametersCall(Scope.kt:284)
	org.koin.core.scope.Scope.resolveInstance(Scope.kt:270)
	org.koin.core.scope.Scope.resolve(Scope.kt:243)
	org.koin.core.scope.Scope.get(Scope.kt:225)
	world.gregs.voidps.engine.EngineModulesKt.engineModule$lambda$0$0(EngineModules.kt:1302)
	org.koin.core.instance.InstanceFactory.create(InstanceFactory.kt:49)
	org.koin.core.instance.SingleInstanceFactory.create(SingleInstanceFactory.kt:46)
	org.koin.core.instance.SingleInstanceFactory.get$lambda$0(SingleInstanceFactory.kt:55)
	org.koin.mp.KoinPlatformTools.synchronized(KoinPlatformTools.kt:36)
	org.koin.core.instance.SingleInstanceFactory.get(SingleInstanceFactory.kt:53)
	org.koin.core.registry.InstanceRegistry.resolveInstance$koin_core(InstanceRegistry.kt:132)
	org.koin.core.resolution.CoreResolver.resolveFromRegistry(CoreResolver.kt:87)
	org.koin.core.resolution.CoreResolver.resolveFromContextOrNull(CoreResolver.kt:74)
	org.koin.core.resolution.CoreResolver.resolveFromContextOrNull$default(CoreResolver.kt:72)
	org.koin.core.resolution.CoreResolver.resolveFromContext(CoreResolver.kt:69)
	org.koin.core.scope.Scope.resolveFromContext(Scope.kt:321)
	org.koin.core.scope.Scope.stackParametersCall(Scope.kt:284)
	org.koin.core.scope.Scope.resolveInstance(Scope.kt:270)
	org.koin.core.scope.Scope.resolve(Scope.kt:243)
	org.koin.core.scope.Scope.get(Scope.kt:225)
	GameTickKt.getTickStages$default(GameTick.kt:128)
	Main.main(Main.kt:68)
[31m[0;39mException in thread "main" org.koin.core.error.InstanceCreationException: Could not create instance for '[Singleton: 'world.gregs.voidps.engine.entity.character.npc.NPCs']'
	at org.koin.core.instance.InstanceFactory.create(InstanceFactory.kt:56)
	at org.koin.core.instance.SingleInstanceFactory.create(SingleInstanceFactory.kt:46)
	at org.koin.core.instance.SingleInstanceFactory.get$lambda$0(SingleInstanceFactory.kt:55)
	at org.koin.mp.KoinPlatformTools.synchronized(KoinPlatformTools.kt:36)
	at org.koin.core.instance.SingleInstanceFactory.get(SingleInstanceFactory.kt:53)
	at org.koin.core.registry.InstanceRegistry.resolveInstance$koin_core(InstanceRegistry.kt:132)
	at org.koin.core.resolution.CoreResolver.resolveFromRegistry(CoreResolver.kt:87)
	at org.koin.core.resolution.CoreResolver.resolveFromContextOrNull(CoreResolver.kt:74)
	at org.koin.core.resolution.CoreResolver.resolveFromContextOrNull$default(CoreResolver.kt:72)
	at org.koin.core.resolution.CoreResolver.resolveFromContext(CoreResolver.kt:69)
	at org.koin.core.scope.Scope.resolveFromContext(Scope.kt:321)
	at org.koin.core.scope.Scope.stackParametersCall(Scope.kt:284)
	at org.koin.core.scope.Scope.resolveInstance(Scope.kt:270)
	at org.koin.core.scope.Scope.resolve(Scope.kt:243)
	at org.koin.core.scope.Scope.get(Scope.kt:225)
	at GameTickKt.getTickStages$default(GameTick.kt:128)
	at Main.main(Main.kt:68)
Caused by: org.koin.core.error.InstanceCreationException: Could not create instance for '[Singleton: 'world.gregs.voidps.engine.data.definition.NPCDefinitions']'
	at org.koin.core.instance.InstanceFactory.create(InstanceFactory.kt:56)
	at org.koin.core.instance.SingleInstanceFactory.create(SingleInstanceFactory.kt:46)
	at org.koin.core.instance.SingleInstanceFactory.get$lambda$0(SingleInstanceFactory.kt:55)
	at org.koin.mp.KoinPlatformTools.synchronized(KoinPlatformTools.kt:36)
	at org.koin.core.instance.SingleInstanceFactory.get(SingleInstanceFactory.kt:53)
	at org.koin.core.registry.InstanceRegistry.resolveInstance$koin_core(InstanceRegistry.kt:132)
	at org.koin.core.resolution.CoreResolver.resolveFromRegistry(CoreResolver.kt:87)
	at org.koin.core.resolution.CoreResolver.resolveFromContextOrNull(CoreResolver.kt:74)
	at org.koin.core.resolution.CoreResolver.resolveFromContextOrNull$default(CoreResolver.kt:72)
	at org.koin.core.resolution.CoreResolver.resolveFromContext(CoreResolver.kt:69)
	at org.koin.core.scope.Scope.resolveFromContext(Scope.kt:321)
	at org.koin.core.scope.Scope.stackParametersCall(Scope.kt:284)
	at org.koin.core.scope.Scope.resolveInstance(Scope.kt:270)
	at org.koin.core.scope.Scope.resolve(Scope.kt:243)
	at org.koin.core.scope.Scope.get(Scope.kt:225)
	at world.gregs.voidps.engine.EngineModulesKt.engineModule$lambda$0$0(EngineModules.kt:1302)
	at org.koin.core.instance.InstanceFactory.create(InstanceFactory.kt:49)
	... 16 more
Caused by: java.lang.IllegalArgumentException: Cannot find npc id to clone 'tz_kih'
	at world.gregs.voidps.engine.data.definition.NPCDefinitions.load$lambda$0$0(NPCDefinitions.kt:42)
	at world.gregs.config.Config.fileReader(Config.kt:11)
	at world.gregs.voidps.engine.data.definition.NPCDefinitions.load$lambda$0(NPCDefinitions.kt:32)
	at world.gregs.voidps.engine.TimedLoaderKt.timedLoad(TimedLoader.kt:18)
	at world.gregs.voidps.engine.data.definition.NPCDefinitions.load(NPCDefinitions.kt:28)
	at Main.cache$lambda$0$3(Main.kt:119)
	at org.koin.core.instance.InstanceFactory.create(InstanceFactory.kt:49)
	... 32 more

</code>
</details>

I checked and the image does include a copy of the cache files which matches the latest version found in the download link in the readme.

**To Reproduce**
Steps to reproduce the behavior:
1. Launch Docker container with `docker container run --rm --interactive --tty greghib/void:2.5.0`
2. Server container should immediately crash

**Expected behavior**
Server container should start the game engine correctly and allow players to connect.

**Screenshots**
Included stacktrace above.

**Desktop (please complete the following information):**
- OS: Ubuntu and Arch

**Additional context**
I was going to try and find another copy of the cache file for this version somewhere else but the readme for Void mentions that the cache has been modified in some way but I'm not sure how.

I tested the 2.4.0 version of the image and the server started as expected.

--- a/data/minigame/tzhaar_fight_pits/tzhaar_fight_pit.npcs.toml
+++ b/data/minigame/tzhaar_fight_pits/tzhaar_fight_pit.npcs.toml
@@ -14,7 +14,7 @@ categories = ["tz_haar"]
 examine = "Some kind of bat..."
 
 [tz_kih_fight_pits_2]
-clone = "tz_kih"
+clone = "tz_kih_fight_pits"
 id = 2628
 
 [tz_kek_fight_pits]

--- a/engine/src/main/kotlin/world/gregs/voidps/engine/data/definition/NPCDefinitions.kt
+++ b/engine/src/main/kotlin/world/gregs/voidps/engine/data/definition/NPCDefinitions.kt
@@ -51,7 +51,7 @@ object NPCDefinitions : DefinitionsDecoder<NPCDefinition> {
                                 "clone" -> {
                                     val name = string()
                                     val npc = refs.getInt(name)
-                                    require(npc >= 0) { "Cannot find npc id to clone '$name'. Make sure it's in the same file." }
+                                    require(npc >= 0) { "Cannot find npc to clone with id '$name' in ${path}. Make sure it's in the same file." }
                                     val definition = definitions[npc]
                                     extras.putAll(definition.extras ?: continue)
                                 }

--- a/engine/src/main/kotlin/world/gregs/voidps/engine/data/definition/NPCDefinitions.kt
+++ b/engine/src/main/kotlin/world/gregs/voidps/engine/data/definition/NPCDefinitions.kt
@@ -37,8 +37,10 @@ object NPCDefinitions : DefinitionsDecoder<NPCDefinition> {
     ) {
         timedLoad("npc extra") {
             val ids = Object2IntOpenHashMap<String>()
+            val refs = Object2IntOpenHashMap<String>()
             ids.defaultReturnValue(-1)
             for (path in paths) {
+                refs.clear()
                 Config.fileReader(path, 150) {
                     while (nextSection()) {
                         val stringId = section()
@@ -48,8 +50,8 @@ object NPCDefinitions : DefinitionsDecoder<NPCDefinition> {
                             when (val key = key()) {
                                 "clone" -> {
                                     val name = string()
-                                    val npc = ids.getInt(name)
-                                    require(npc >= 0) { "Cannot find npc id to clone '$name'" }
+                                    val npc = refs.getInt(name)
+                                    require(npc >= 0) { "Cannot find npc id to clone '$name'. Make sure it's in the same file." }
                                     val definition = definitions[npc]
                                     extras.putAll(definition.extras ?: continue)
                                 }
@@ -80,6 +82,7 @@ object NPCDefinitions : DefinitionsDecoder<NPCDefinition> {
                             }
                         }
                         require(!ids.containsKey(stringId)) { "Duplicate npc id found '$stringId' at $path." }
+                        refs[stringId] = id
                         ids[stringId] = id
                         definitions[id].stringId = stringId
                         if (extras.isNotEmpty()) {

--- a/engine/src/main/kotlin/world/gregs/voidps/engine/data/definition/ObjectDefinitions.kt
+++ b/engine/src/main/kotlin/world/gregs/voidps/engine/data/definition/ObjectDefinitions.kt
@@ -35,8 +35,10 @@ object ObjectDefinitions : DefinitionsDecoder<ObjectDefinition> {
     fun load(paths: List<String>): ObjectDefinitions {
         timedLoad("object extra") {
             val ids = Object2IntOpenHashMap<String>()
+            val refs = Object2IntOpenHashMap<String>()
             ids.defaultReturnValue(-1)
             for (path in paths) {
+                refs.clear()
                 Config.fileReader(path) {
                     while (nextSection()) {
                         val stringId = section()
@@ -50,8 +52,8 @@ object ObjectDefinitions : DefinitionsDecoder<ObjectDefinition> {
                                 "mining" -> extras[key] = Rock(this)
                                 "clone" -> {
                                     val name = string()
-                                    val obj = ids.getInt(name)
-                                    require(obj >= 0) { "Cannot find object id to clone '$name'" }
+                                    val obj = refs.getInt(name)
+                                    require(obj >= 0) { "Cannot find object to clone with id '$name' in ${path}. Make sure it's in the same file." }
                                     val definition = definitions[obj]
                                     extras.putAll(definition.extras ?: continue)
                                 }
@@ -67,6 +69,7 @@ object ObjectDefinitions : DefinitionsDecoder<ObjectDefinition> {
                         }
                         require(!ids.containsKey(stringId)) { "Duplicate object id found '$stringId' at $path." }
                         ids[stringId] = id
+                        refs[stringId] = id
                         definitions[id].stringId = stringId
                         if (extras.isNotEmpty()) {
                             definitions[id].extras = extras

